taglib/mpeg/id3v2/id3v2framefactory.cpp

```diff
@@ -147,11 +147,9 @@ std::pair<Frame::Header *, bool> FrameFactory::prepareFrameHeader(
   }
 #endif
 
-  for(auto it = frameID.cbegin(); it != frameID.cend(); it++) {
-    if( (*it < 'A' || *it > 'Z') && (*it < '0' || *it > '9') ) {
-      delete header;
-      return {nullptr, false};
-    }
+  if(std::any_of(frameID.begin(), frameID.end(), [](auto c) { return (c < 'A' || c > 'Z') && (c < '0' || c > '9'); })) {
```

Comment:
The original loop uses `cbegin` and `cend`. Is there an advantage to using the const versions?

Reply:
Not really. This piece of code was actually transformed into a for range loop first which is incompatible with cbegin and cend.